src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -49,6 +59,106 @@ public async Task PutJavaScript()
+                            case IndexType.JavaScriptMap:
+                                index = MapIndex.CreateNewForTest(indexDefinition, context);
+                                break;
+                            case IndexType.MapReduce:
+                            case IndexType.JavaScriptMapReduce:
+                                index = MapReduceIndex.CreateNewForTest<MapReduceIndex>(indexDefinition, context.DocumentDatabase, context);
+                                break;
+                            default:
+                                throw new Exception();
+                        }
+                        break;
+                    case IndexSourceType.Counters:
+                        switch (indexDefinition.Type)
+                        {
+                            case IndexType.Map:
+                            case IndexType.JavaScriptMap:
+                                index = MapCountersIndex.CreateNewForTest(indexDefinition, context);
+                                break;
+                            case IndexType.MapReduce:
+                            case IndexType.JavaScriptMapReduce:
+                                index = MapReduceIndex.CreateNewForTest<MapReduceCountersIndex>(indexDefinition, context.DocumentDatabase, context);
+                                break;
+                            default:
+                                throw new Exception();
+                        }
+                        break;
+                    case IndexSourceType.TimeSeries:
+                        switch (indexDefinition.Type)
+                        {
+                            case IndexType.Map:
+                            case IndexType.JavaScriptMap:
+                                index = MapTimeSeriesIndex.CreateNewForTest(indexDefinition, context.DocumentDatabase, context);
+                                break;
+                            case IndexType.MapReduce:
+                            case IndexType.JavaScriptMapReduce:
+                                index = MapReduceIndex.CreateNewForTest<MapReduceTimeSeriesIndex>(indexDefinition, context.DocumentDatabase, context);
+                                break;
+                            default:
+                                throw new Exception(); 
+                        }
+                        break;
+                    default:
+                        throw new Exception();
+                }
+                
+                if (input.TryGet("Query", out BlittableJsonReaderObject query) == false || query is null)
+                {
+                    var queryString = $"from index '{index.Name}'";
+                    var queryObject = new { Query = queryString };
+                    
+                    query = DocumentConventions.DefaultForServer.Serialization.DefaultConverter.ToBlittable(queryObject, context);
+                }
+
+                index.Start();
```

Comment:
Maybe instead having 'InitializeTestIndex' we can pass parameters here? Thoughts?

Reply:
I think we need to dispose this index somewhere? I do not see that code?